kitsune/search/v2/base.py

```diff
@@ -32,13 +36,21 @@ def prepare(cls, instance):
                 # if the field is an Object but the value isn't an InnerDoc
                 # or a list containing an InnerDoc then we're dealing with locales
                 locale = obj.prepare_locale(instance)
-                if locale and value:
+                # Check specifically against None, False is a valid value
+                if locale and (value is not None):
                     obj[f] = {locale: value}
+
             else:
+                # if merge_docs is True, then we know that we don't have
+                # a SumoLocaleAware field here. We need to get the value from the parent
+                if merge_docs and instance.parent:
+                    value = obj.get_field_value(f, instance.parent, prepare_method)
```

Comment:
This is nice, I think the comment should read more like: "if merge_docs is True, **and** we know that we don't have..."